sia/miner/lock.go

```diff
@@ -0,0 +1,21 @@
+package miner
+
+// lock write-locks the miner.
+func (m *Miner) lock() {
+	m.rwLock.Lock()
+}
+
+// unlock write-unlocks the miner.
+func (m *Miner) unlock() {
+	m.rwLock.Unlock()
+}
+
+// rLock read-locks the miner.
+func (m *Miner) rLock() {
+	m.rwLock.RLock()
+}
+
+// rUnlock read-unlocks the miner.
+func (m *Miner) rUnlock() {
+	m.rwLock.RUnlock()
+}
```

Comment:
I'm starting to wonder if there's a good reason to do this, but I can't think of one.